Test 2: Add test function to test edit task and clear completed functions

ookino opened the pull request:
## Features Added ♻️ 
- Added test function to test editing a task
- Added test function to test marking a task as completed
- Added test function to test clearing all completed tasks

ookino committed:
Add test function to to test edit task and clear completed functions

henatan99 commented on src/test/task.test.js:
```diff
@@ -29,3 +33,61 @@ describe('Remove Task', () => {
+  it('check length of array after removing completed tasks', () => {
+    const checkCompleted = clearCompleted(removeCompleted);
+    expect(checkCompleted).toHaveLength(1);
+  });
```
- [ ] You did excellent work :clap:. I think you can also add a test for the function for updating an item's index value upon item deletion or dragging if applicable.  :pray:

yaseerokino replied:
@henatan99 thank you for the review, but the drag and drop function was optional and I did not implement it in my project. also when an item is deleted the index do not update, rather what happens is when adding a new task the function checks the existing tasks for the max index number and adds one to the max index number to create a new index

henatan99 replied:
Hi, @yaseerokino, I said the dragging only if applicable. But at least the app should update the item's index when an item is deleted. For example, if you have items with index: 1, index: 2, and index: 3, and if you delete an item with index: 2, then an item with index: 3, should be updated to index: 2.  Therefore , you need to add a test if the indexes are updated after the list is changed. :pray:

yaseerokino replied:
i have made the required changes

henatan99 reviewed (changes requested):
Hi @yaseerokino , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ookino committed:
Update previous tests

ookino committed:
Add update index function test

Added a function to test if the array's tasks index update after an index has been deleted

oxenprogrammer reviewed (approved):
Hi @yaseerokino , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

**Optional change**
- [ ] The package lodash is included as both a dev and production dependency. Please remove it from core dependencies. 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
